.github/workflows/ci.yml

```diff
@@ -49,8 +51,14 @@ jobs:
     - name: Test
       run: mage CITest
 
-    - name: Generate Swagger
-      run: mage spec
+    - name: Update spec and push generated spec file
+      if: github.event_name == 'pull_request' && github.event.action != 'closed' && github.event.pull_request.merged != true
+      run: |
+        mage spec
```

Comment:
`mage spec` is also doing formatting of files. See https://github.com/TBD54566975/ssi-service/blob/7047be55d3d57ca5e4ee4919126e4acdeb275444/magefile.go#L114-L117

I think that needs to be broken up into separate mage targets. 

As it stands, you might get a passing CI during branch review, but failing in main because of this (a linter might fail).

Reply:
done